src/controllers/redirect.controller.ts

```diff
@@ -28,27 +35,31 @@ export class RedirectController extends BaseController {
     const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
     const token: string = this.session.getAccessToken(this.httpContext.request)
     const dissolution: Optional<DissolutionGetResponse> = await this.service.getDissolution(token, dissolutionSession)
+    const userEmail: string = this.session.getUserEmail(this.httpContext.request)
 
     if (!dissolution) {
       return this.redirect(SELECT_DIRECTOR_URI)
     }
 
-    const signingDirector: Optional<DissolutionGetDirector> = this.getUserPendingSignature(dissolution!)
-
-    if (dissolution!.application_status === ApplicationStatus.PENDING_APPROVAL && signingDirector) {
+    const isApplicant: boolean = dissolution!.created_by === userEmail
+    const signingDirector: Optional<DissolutionGetDirector> = this.getUserPendingSignature(dissolution!, userEmail)
 
-      dissolutionSession.approval = this.prepareApprovalData(dissolution!, signingDirector)
-      this.session.setDissolutionSession(this.httpContext.request, dissolutionSession)
+    if (dissolution!.application_status === ApplicationStatus.PENDING_APPROVAL) {
+      if (signingDirector) {
+        dissolutionSession.approval = this.prepareApprovalData(dissolution!, signingDirector)
+        this.session.setDissolutionSession(this.httpContext.request, dissolutionSession)
 
-      return this.redirect(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
+        return this.redirect(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
+      } else if (isApplicant) {
+        return this.redirect(WAIT_FOR_OTHERS_TO_SIGN_URI)
+      } else {
+        return this.redirect(CERTIFICATE_SIGNED_URI)
+      }
```

Comment:
Can we move this into a private `handlePendingApprovalRedirect` private function or something? Just to keep the public method as lean and readable as possible as we will be extending it over time.